tests/unit/test_cli.py

```diff
@@ -7,15 +7,26 @@
 from cou.steps import UpgradeStep
 
 
-def test_parse_args():
-    args = ["--dry-run", "--log-level", "DEBUG", "--interactive", "true"]
+@pytest.mark.parametrize(
+    "args, expected_results",
+    [
+        (["--run", "--log-level", "DEBUG", "--non-interactive"], [True, "DEBUG", True]),
+        (["--run", "--log-level", "warning"], [True, "WARNING", False]),
+        (["--log-level", "debug"], [False, "DEBUG", False]),
+    ],
+)
+def test_parse_args(args, expected_results):
     parsed_args = parse_args(args)
-    assert parsed_args.dry_run
-    assert parsed_args.loglevel == "DEBUG"
-    assert parsed_args.interactive
+    expected_run = expected_results[0]
+    expected_loglevel = expected_results[1]
+    expected_non_interactive = expected_results[2]
+
+    assert parsed_args.run == expected_run
+    assert parsed_args.loglevel == expected_loglevel
+    assert parsed_args.non_interactive == expected_non_interactive
 
     with pytest.raises(ArgumentError):
-        args = parse_args(["--dry-run", "--log-level=DDD"])
+        args = parse_args(["--log-level=DDD"])
```

Comment:
This should be in different test. it's not good practice test different behavior in one test and even worse here we are testing this in each parametrize run (3 times).

Reply:
Solved